content/docs/configuration-services.md

```diff
@@ -0,0 +1,327 @@
+#### ReferenceFinder
+The `ReferenceFinder` service is responsible for handling a *Find References Request*. This service is used to find all references to a given symbol inside of a document.
+
+*Dependencies:* [NameProvider](#nameprovider) | [References](#references)#### GoToResolver
+#### DocumentHighlighter
+The `DocumentHighlighter` service is responsible for handling a *Document Highlights Request*. This service will find all references to a symbol at a given position (via the `References` service) and highlight all these references in a given document.
+
+*Dependencies:* [References](#references) | [NameProvider](#nameprovider)
+#### RenameHandler
+The `RenameHandler` service is responsible for handling a *Rename Request* or a *Prepare Rename Request*. First, the service will check the validity of the *Prepare Rename Request*. If the request is valid, the service will find all references to the selected symbol inside of a document and replace all occurences with the new value.
+
+*Dependencies:* [ReferenceFinder](#referencefinder) | [References](#references) | [NameProvider](#nameprovider)
+#### AstNodeLocator
+The `AstNodeLocator` is responsible for finding a particular `AstNode` inside of a document and return its path or the node itself.
+#### AstNodeDescriptionProvider
+The `AstNodeDescriptionProvider` is responsible for creating a description for a given `AstNode`. By default the description includes the `node`, its `name`, `type`, and `path`, as well as the `documentUri` of the document where the `AstNode` is located.
+
+*Dependencies:* [AstNodeLocator](#astnodelocator) | [NameProvider](#nameprovider)
+#### ReferenceDescriptionProvider
+The `ReferenceDescriptionProvider` is responsible for creating a description for a given reference. By default the description includes the `sourceUri`, `sourcePath`, `targetUri`, `targetPath`, `segment`, and if the reference is `local`.
+
+*Dependencies:* [Linker](#linker) | [AstNodeLocator](#astnodelocator)
+#### Linker
+The `Linker` service is responsible for resolving cross-references in the AST.
+
+*Dependencies:* [AstReflection](#astreflection) | [LangiumDocuments](#langiumdocuments) | [ScopeProvider](#scopeprovider) | [AstNodeLocator](#astnodelocator)
+#### NameProvider
+The `NameProvider` service is responsible for getting the name of a given `AstNode`. If the `AstNode` does not have a name, the `NameProvider` returns `undefined`.
+#### ScopeProvider
+
+*Dependencies:* [AstReflection](#astreflection) | [IndexManager](#indexmanager)
+#### ScopeComputation
+
+*Dependencies:* [NameProvider](#nameprovider) | [AstNodeDescriptionProvider](#astnodedescriptionprovider)
+#### References
+The `References` service is responsible for finding all references to the target node.
+
+*Dependencies:* [NameProvider](#nameprovider) | [IndexManager](#indexmanager) | [AstNodeLocator](#astnodelocator)
+#### JsonSerializer
+The `JsonSerializer` service is responsible for serializing and deserializing JSON.
```

Comment:
Is that any easy to see use case for that?

Reply:
We use it in Langium to serialize and parse the grammar. This might also be useful to export ASTs to other tools or to the IDE client.